### ISSUE TITLE
SingleTileImageryProvider crossing 180 degrees longitude does not render

### ISSUE THREAD

**Issue by mramato** [AnalyticalGraphicsInc/cesium#2195]
I'm guessing this is a more general problem than with just `SingleTileImageryProvider`, but it's the easiest way to reproduce the issue.

__Sandcastle example__
```
var viewer = new Cesium.Viewer('cesiumContainer', {
    baseLayerPicker : false
});
 
viewer.scene.imageryLayers.addImageryProvider(new Cesium.SingleTileImageryProvider({
    url : '../images/Cesium_Logo_overlay.png',
    rectangle : Cesium.Rectangle.fromDegrees(100, 60, -179, 78)
}));
```

**Comment by bagnell** [comment]
This is broken again. `Rectangle.intersection` does not handle intersection with `Rectangle.MAX_VALUE` properly. Also `ImageryLayer._createTileImagerySkeletons` needs to not assume that `west` is strictly less than `east` for all rectangles.

**Comment by mramato** [comment]
Could the MAX_VALUE issue be related to #2624?

**Comment by bagnell** [comment]
I don't think so. I believe that issue is with `Cesium.Math.convertLongitudeRange` and this one is definitely with `Rectangle.intersection`.
